crates/multi-transaction-payment/src/lib.rs

```diff
@@ -0,0 +1,169 @@
+// #![deny(warnings)]
+#![cfg_attr(test, feature(proc_macro_hygiene))]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use dex_general::{ExportDexGeneral, WeightInfo};
+use frame_support::{
+    ensure,
+    traits::{Currency, Get, IsSubType, OnUnbalanced},
+};
+pub use pallet::*;
+use pallet_transaction_payment::OnChargeTransaction;
+use sp_runtime::{
+    traits::{DispatchInfoOf, PostDispatchInfoOf, Zero},
+    transaction_validity::{InvalidTransaction, TransactionValidityError},
+};
+use sp_std::{boxed::Box, vec::Vec};
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+type CallOf<T> = <T as Config>::RuntimeCall;
+type SubstrateDefaultPayment<T> =
+    pallet_transaction_payment::CurrencyAdapter<<T as Config>::Currency, <T as Config>::OnUnbalanced>;
+
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+    use frame_support::{
+        dispatch::{DispatchInfo, GetDispatchInfo, PostDispatchInfo},
+        pallet_prelude::*,
+    };
+    use frame_system::pallet_prelude::*;
+    use sp_runtime::traits::Dispatchable;
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config<RuntimeCall = CallOf<Self>> + currency::Config<Balance = u128> {
+        /// The aggregated call type.
+        type RuntimeCall: Parameter
+            // + Dispatchable<RuntimeOrigin = Self::RuntimeOrigin, PostInfo = PostDispatchInfo, Info = DispatchInfo>
+            + Dispatchable<RuntimeOrigin = Self::RuntimeOrigin, PostInfo = PostDispatchInfo, Info = DispatchInfo>
+            + GetDispatchInfo
+            + IsSubType<Call<Self>>;
+
+        /// dex api
+        type Dex: ExportDexGeneral<Self::AccountId, Self::CurrencyId>;
+
+        /// The currency trait.
+        type Currency: Currency<Self::AccountId, Balance = u128>;
+
+        /// What to do with fees. Typically this is a transfer to the block author or treasury
+        type OnUnbalanced: OnUnbalanced<<Self::Currency as Currency<Self::AccountId>>::NegativeImbalance>;
+
+        /// weights of dex operations
+        type DexWeightInfo: dex_general::WeightInfo;
+    }
+
+    #[pallet::error]
+    pub enum Error<T> {}
+
+    #[pallet::pallet]
+    pub struct Pallet<T>(_);
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        #[pallet::call_index(0)]
+        #[pallet::weight({
+			let dispatch_info = call.get_dispatch_info();
+			((T::DexWeightInfo::swap_assets_for_exact_assets(_path.len() as u32)).saturating_add(dispatch_info.weight), dispatch_info.class,)
+		})]
+        #[frame_support::transactional]
+        pub fn with_fee_swap_path(
+            origin: OriginFor<T>,
+            _path: Vec<T::CurrencyId>,
```

Comment:
Any reason not to use a `BoundedVec` here? Otherwise the logic for checking the swap path could be a source of a DoS attack

Reply:
It's a pass-through to the dex-general swapping function, which is using a Vec rather than a boundedVec. It shouldn't be a DOS vulnerability since the weight of the dex swapping function is parametrized by the vector length